stage1/reaper/reaper.sh

```diff
@@ -2,17 +2,31 @@
 shopt -s nullglob
 
 SYSCTL=/usr/bin/systemctl
+DIAG=/diagnostic
 
 if [ $# -eq 1 ]; then
     app=$1
     status=$(${SYSCTL} show --property ExecMainStatus "${app}.service")
-    echo "${status#*=}" > "/rkt/status/$app"
-    if [ "${status#*=}" != 0 ] ; then
+    status="${status#*=}"
+    echo "${status}" > "/rkt/status/$app"
+    if [ "${status}" != 0 ] ; then
         # The command "systemctl exit $status" sets the return value that will
         # be used when the pod exits (via shutdown.service).
         # This command is available since systemd v227. On older versions, the
         # command will fail and rkt will just exit with return code 0.
         ${SYSCTL} exit ${status#*=} 2>/dev/null
+        # systemd EXIT_EXEC is 203 (https://github.com/systemd/systemd/blob/v230/src/basic/exit-status.h#L47)
+        if [ "${status}" == 203 ]; then
+            # Run diagnostic
+            unit_file=$(${SYSCTL} cat "${app}.service")
+            target=${unit_file##*ExecStart=}
+            target=${target%%RootDirectory=*}
+            target=${target#'"'*}   # remove left quote
+            target=${target%'"'*}   # remove right quote
+            root_dir=$(${SYSCTL} show --property RootDirectory "${app}.service")
+            root_dir=${root_dir#*=}
```

Comment:
About the use of `--property` and `--value`:
```
$ systemctl show --property ExecStart --value dhcpcd.servic

{ path=/usr/bin/dhcpcd ; argv[]=/usr/bin/dhcpcd -q -b ; ignore_errors=no ; start_time=[Wed 2016-07-06 11:29:43 CEST] ; stop_time=[Wed 2016-07-06 11:29:43 CEST] ; pid=270 ; code=exited ; status=0 }
```
Instead
```
unit_file=systemctl cat dhcpcd.service
target=${unit_file##*ExecStart=}
target=${target%%RootDirectory=*}
```
Result in
```
 "/usr/bin/dhcpcd -q -b"
```
Also the diagnostic ignore arguments so it needs only the binary